crates/extrinsics/src/lib.rs

```diff
@@ -310,17 +347,29 @@ pub fn parse_code_hash(input: &str) -> Result<<DefaultConfig as Config>::Hash> {
     Ok(arr.into())
 }
 
+/// Fetch the hash of the *best* block (included but not guaranteed to be finalized).
+async fn get_best_block(
+    rpc: RpcClient,
```

Comment:
maybe we should align each function to use LegacyRpc `rpc: &LegacyRpcMethods::<DefaultConfig>`